public/app/plugins/datasource/prometheus/legend.test.ts

```diff
@@ -23,8 +23,8 @@ describe('renderLegendFormat()', () => {
   it('Bad syntax', () => {
     expect(renderLegendFormat('value: {{a}', labels)).toEqual('value: {{a}');
     expect(renderLegendFormat('value: {a}}}', labels)).toEqual('value: {a}}}');
-
-    // Current behavior -- not sure if expected or not
-    expect(renderLegendFormat('value: {{{a}}}', labels)).toEqual('value: {a}');
+    expect(renderLegendFormat('{{}}', labels)).toEqual('{{}}');
+    expect(renderLegendFormat('{{{{a', labels)).toEqual('{{{{a');
+    expect(renderLegendFormat('value: {{{a}}}', labels)).toEqual('value: {{{a}}}');
```

Comment:
This is a change in behavior from the previous syntax, but "fixing" it is too weird --  I don't think `{a}` is really a better option

Reply:
is there a reason not to just change the regex to something like `/\{\{\s*([^{}]+?)\s*\}\}/g;` ?

Reply:
i.e. do we support replacement strings with curly braces inside them?

Reply:
the reason for the change in behavior with that regex (which is the same as proposed in #43054 by @leeoniya) is that the current regexp interprets `{{{a}}}` as `[{{] [{a] [}}] }` and does not replace it because `{a` is not a valid variable name, whereas the updated regex interprets it (correctly imho) as `{ [{{] [a] [}}] }` and replaces `{{a}}` with the value of variable `a`.